src/LandIce/evaluators/LandIce_StokesFOBasalResid_Def.hpp

```diff
@@ -113,19 +103,24 @@ operator() (const StokesFOBasalResid_Tag& tag, const int& sideSet_idx) const {
   const int cell = sideSet.elem_LID(sideSet_idx);
   const int side = sideSet.side_local_id(sideSet_idx);
 
-  const ScalarT ff = (regularized) ? pow(10.0, -10.0*homotopyParam(0)) : ScalarT(0);
-
   ScalarT local_res[2];
 
   for (unsigned int node=0; node<numSideNodes; ++node) {
     local_res[0] = 0.0;
     local_res[1] = 0.0;
-    for (unsigned int dim=0; dim<vecDimFO; ++dim) {
-      for (unsigned int qp=0; qp<numSideQPs; ++qp) {
-        local_res[dim] += (ff + beta(sideSet_idx,qp)*u(sideSet_idx,qp,dim))*BF(sideSet_idx,node,qp)*w_measure(sideSet_idx,qp);
-      }
-      Kokkos::atomic_add(&residual(cell,sideNodes(side,node),dim), local_res[dim]);
+
+    for (unsigned int qp=0; qp<numSideQPs; ++qp) {
+      MeshScalarT bx = -normals(sideSet_idx,qp,0)/normals(sideSet_idx,qp,2);
+      MeshScalarT by = -normals(sideSet_idx,qp,1)/normals(sideSet_idx,qp,2);
+      // removed the following line after checking performance
+      bx=0.0; by=0.0;
```

Comment:
Remove?

Reply:
I'd like to check whether requiring the normal computations significantly impacts the performances. Removing that line will make several tests fail and we'll have reset the regression results.

Reply:
Ah okay got it.